cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -2611,6 +2611,107 @@ TEST_F(ScalarTemporalTest, TestRoundTemporal) {
   CheckScalarUnary(op, unit, times, unit, round_15_years, &round_to_15_years);
 }
 
+TEST_F(ScalarTemporalTest, TestCeilFloorRoundTemporalAmbiguous1) {
+  auto unit = timestamp(TimeUnit::MILLI, "Asia/Tehran");
+  auto options = RoundTemporalOptions(1, CalendarUnit::HOUR, /*some_parameter=*/true);
+  // Asia/Tehran switched from UTC+X to UTC+Y on 2022-03-31 HH:mm:ss
```

Comment:
The intent was that you fill the actual parameter name, and timezone/timestamps values :-)

Reply:
So far I have something like this:
```
  // Asia/Tehran switches from UTC+3:30 to UTC+4:30 on 2022-03-22 00:00:00 UTC+3:30
  // This causes an hour long non-existing period in local time.
```